nodedb.go

```diff
@@ -228,6 +262,12 @@ func (ndb *nodeDB) SaveOrphans(version int64, orphans map[string]int64) {
 	ndb.mtx.Lock()
 	defer ndb.mtx.Unlock()
 
+	//var toVersion int64
```

Comment:
why is this commented out?

Reply:
When saving the previous version is always `version - 1` as you must have been on a version when you hit save, I did not see the need to look up in database what you already know. So I refactored here; for other solutions refactoring was necessary as versions where in memory not in a database hence the tendermint solution would not work.
With `hack` solution, they are in a database so could use a modified version of the original tendermint code.